sql/executor.go

```diff
@@ -429,6 +439,56 @@ func (p parameters) Arg(name string) (parser.Datum, bool) {
 	}
 }
 
+var _ parser.Args = golangParameters{}
+
+type golangParameters []interface{}
```

Comment:
This is the problem with a PR like this, which adds plumbing but not the functionality that depends on it. Without seeing how it is used, how can the ergonomics of this interface be reasonably evaluated?